src/callbacks_step/amr.jl

```diff
@@ -346,6 +356,168 @@ function (amr_callback::AMRCallback)(u_ode::AbstractVector, mesh::TreeMesh,
     return has_changed
 end
 
+# `passive_args` is currently used for Euler with self-gravity to adapt the gravity solver
+# passively without querying its indicator, based on the assumption that both solvers use
+# the same mesh. That's a hack and should be improved in the future once we have more examples
+# and a better understanding of such a coupling.
+# `passive_args` is expected to be an iterable of `Tuple`s of the form
+# `(p_u_ode, p_mesh, p_equations, p_dg, p_cache)`.
+function (amr_callback::AMRCallback)(u_ode::AbstractVector, mesh::TreeMesh,
+                                     equations, dg::DG,
+                                     cache, cache_parabolic,
+                                     semi::SemidiscretizationHyperbolicParabolic,
+                                     t, iter;
+                                     only_refine = false, only_coarsen = false,
+                                     passive_args = ())
+    @unpack controller, adaptor = amr_callback
+
+    u = wrap_array(u_ode, mesh, equations, dg, cache)
+    # TODO: Keep indicator based on hyperbolic variables?
+    lambda = @trixi_timeit timer() "indicator" controller(u, mesh, equations, dg, cache,
+                                                          t = t, iter = iter)
+
+    if mpi_isparallel()
+        # Collect lambda for all elements
+        lambda_global = Vector{eltype(lambda)}(undef, nelementsglobal(dg, cache))
+        # Use parent because n_elements_by_rank is an OffsetArray
+        recvbuf = MPI.VBuffer(lambda_global, parent(cache.mpi_cache.n_elements_by_rank))
+        MPI.Allgatherv!(lambda, recvbuf, mpi_comm())
+        lambda = lambda_global
+    end
```

Comment:
You're right, my bad. In that case it might be nice to leave it in there for future use. However, I'd recommend to comment out the implementation inside the `if` condition and add an `error("MPI has not been verified yet")` or similar, such that someone who tries to use this for the first time in 2D does not assume that everything is tested etc.